Lib/Hash.hpp

```diff
@@ -215,6 +237,17 @@ class DefaultHash
     return hash;
   }
 
+  template<class Iter>
+  static unsigned hashIter(
+      Iter iter,
+      unsigned hash = FNV32_OFFSET_BASIS
+      ) {
+    while (iter.hasNext()) {
+      hash = (hash ^ iter.next()) * FNV32_PRIME;
```

Comment:
Is this still an open thing?

Reply:
ahhh right. Tbh i didn't really care how exactly we hash. We can refactor this as you said to remove `HashUtils::combine` but I think this will be a bit of effort so I'm not so keen on it right now. So I'd say let's defer this to a separate PR?